src/join.rs

```diff
@@ -77,7 +78,13 @@ pub trait Join {
     {
         JoinIter::new(self)
     }
-
+    /// Create a joined parallel iterator over the contents.
+    #[cfg(feature="parallel")]
+    fn par_join(self) -> JoinParIter<Self>
```

Comment:
Hmmm.... So I should really be creating new trait that has that? That would propably resolve some problems I have had. Have to look into it.

Currently all storages need to support random indexing (Join::get) and only time splitting then be impossible would be if the structure needed to touch mutably other indices when indexing or if distinct indices would return mutable references to same index.